boa/test/plugin.py

```diff
@@ -1,22 +1,40 @@
+import contextlib
 from typing import Generator
 
 import hypothesis
 import pytest
 
 import boa
+from boa.profiling import print_call_profile
 
 # monkey patch HypothesisHandle. this fixes underlying isolation for
 # hypothesis.given() and also hypothesis stateful functionality.
 _old_init = hypothesis.core.HypothesisHandle.__init__
 
 
+@contextlib.contextmanager
+def _enable_profiling(flag: bool):
+    try:
+        boa.env._profile_calls = flag
+        yield
+    finally:
+        boa.env._profile_calls = False
```

Comment:
this has an interesting effect of disabling `_profile_calls` even if it was set to `True` before `_enable_profiling()` was called.

Reply:
renamed it to `_toggle_profiling`